### PR TITLE
added map and location on public event page

### LINKED ISSUE #14: Display the event location on a map and show the event address in the public event page
screen shot 2017-04-23 at 10 08 45 pm

**What is expected ?**
- Map and address as a single [component](https://guides.emberjs.com/v2.12.0/getting-started/core-concepts/#toc_components) added to the public event page (the component should accept attributes that can allow us to either render just the map or just the address or both side by side.
- The width of it should be fluid. (that is, it should take up the width of it's parent)
- Data provided by the API should be used. (`latitude`, `longitude`, `locationName`)

**References:**
- https://guides.emberjs.com/v2.12.0/components/defining-a-component/
- https://semantic-ui.com/collections/grid.html
- https://github.com/asennikov/ember-g-map

### PR DESCRIPTION
**Pull request by anu-007**
added map and location with the street view two fingers dragging on touchscreen devices on the public event page.

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

#### Short description of what this resolves:
Added location and address support to public event page.

#### Changes proposed in this pull request:

-Added location and address support

Fixes #14

### CODE REVIEW

File: tests/integration/components/public/event-map-test.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import { moduleForComponent, test } from 'ember-qunit';
+import hbs from 'htmlbars-inline-precompile';
+
+moduleForComponent('g-map', 'Integration | Component | g map', {
```
Comment (niranjan94) [inline tests/integration/components/public/event-map-test.js]: See other integration tests to see which `moduleForComponent` method is used and how it's used. Modify accordingly.

Comment (niranjan94) [inline tests/integration/components/public/event-map-test.js]: And your test needs to be for `public/event-map` component. Not `g-map` component. `g-map` is already well tested by the library developer.

**Comment by anu-007** [comment]
@niranjan94 please review changes

File: tests/integration/components/public/event-map-test.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import { test } from 'ember-qunit';
+import moduleForComponent from 'open-event-frontend/tests/helpers/component-helper';
+import hbs from 'htmlbars-inline-precompile';
+
+moduleForComponent('public/event-map', 'Integration | Component | public/event map');
+
+test('it renders', function(assert) {
+  assert.expect(2);
+
+  this.render(hbs`{{g-map}}`);
```
Comment (niranjan94) [inline tests/integration/components/public/event-map-test.js]: Again ... why are you rendering g-map and testing it ? We need to test the event-map component. Not g-map component. I'm sure ember would have generated a proper base test for you. I'm not sure why you decided to remove that.

File: tests/integration/components/public/event-map-test.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  this.render(hbs`
+    {{#g-map}}
+      template block text
+    {{/g-map}}
+  `);
+
+  assert.ok(this.$());
+});
+
+test('it includes child marker', function(assert) {
```
Comment (niranjan94) [inline tests/integration/components/public/event-map-test.js]: Not needed. We're not testing g-map. We want to test `event-map`

**Comment by anu-007** [comment]
@niranjan94 please review changes.

File: app/templates/components/public/event-map.hbs [hunk 4]
```diff
@@ -0,0 +1,11 @@
+<div class="ui grid fluid container">
```
Comment (niranjan94) [inline app/templates/components/public/event-map.hbs]: Please add these classes to the class definition at `event-map.js` and remove this div.

Ref: https://guides.emberjs.com/v2.13.0/components/customizing-a-components-element/

File: app/templates/components/public/event-map.hbs [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+  <div class="eight wide column loc-map">
+    {{#g-map lat=event.latitude lng=event.longitude zoom=12 gestureHandling="cooperative" streetView="StreetViewPanorama" as |context|}}
```
Comment (niranjan94) [inline app/templates/components/public/event-map.hbs]: Use single quotes within ember helpers.

Please see: https://github.com/fossasia/open-event-frontend/blob/development/docs/style-guide.md

**Comment by anu-007** [comment]
@niranjan94 please review changes

File: app/templates/components/public/event-map.hbs [hunk 6]
```diff
@@ -0,0 +1,9 @@
+<div class="eight wide column loc-map">
```
Comment (niranjan94) [inline app/templates/components/public/event-map.hbs]: Please use proper class names `loc-map` doesn't make any sense. You could call it `event-map`

**Comment by anu-007** [comment]
@niranjan94 changed class name to event-map please review

File: app/styles/pages/public-event.scss [hunk 7]
```diff
@@ -104,3 +104,12 @@
     margin-top: 15px;
   }
 }
+
+.loc-map > .g-map {
```
Comment (niranjan94) [inline app/styles/pages/public-event.scss]: This is still not changed. Please pay more attention to detail. Don't miss things like this.

**Comment by anu-007** [comment]
@niranjan94 changed class name in public-event.scss please review.

**Pull request merged**